dataland-internal-storage/src/main/kotlin/db/migration/V6__FlattenEnvironmentalObjectiveMapsInEuTaxonomyForNonFinancials.kt

```diff
@@ -0,0 +1,99 @@
+    private enum class CriteriaType(val prefix: String) {
+        SubstantialContribution("substantialContribution"),
+        Dnsh("dnsh"),
+    }
+
+    override fun migrate(context: Context?) {
+        migrateCompanyAssociatedDataOfDatatype(
+            context,
+            "eutaxonomy-non-financials",
+            this::flattenEnvironmentalObjectiveMaps,
+        )
+    }
+
+    /**
+     * Migrates an old eu taxonomy non financials dataset to the new format
+     */
+    fun flattenEnvironmentalObjectiveMaps(dataTableEntity: DataTableEntity) {
+        val dataset = dataTableEntity.dataJsonObject
+        listOf("revenue", "capex", "opex").forEach {
+            migrateCashFlow(dataset.getOrJavaNull(it) as JSONObject? ?: return@forEach)
+        }
+        dataTableEntity.companyAssociatedData.put("data", dataset.toString())
+    }
+
+    private fun migrateCashFlow(cashFlow: JSONObject) {
+        migrateFieldNames(cashFlow)
+        migrateMapForCriteria(cashFlow, CriteriaType.SubstantialContribution)
+        (cashFlow.getOrJavaNull("alignedActivities") as JSONArray? ?: return).forEach {
+            migrateMapForCriteria(it as JSONObject, CriteriaType.SubstantialContribution)
+            migrateMapForCriteria(it, CriteriaType.Dnsh)
+        }
+    }
+
+    private fun migrateMapForCriteria(baseObject: JSONObject, criteriaType: CriteriaType) {
+        val fieldRenamingHelper = mapOf(
+            "ClimateMitigation" to "ClimateChangeMitigation",
+            "ClimateAdaptation" to "ClimateChangeAdaption",
+            "Water" to "SustainableUseAndProtectionOfWaterAndMarineResources",
+            "CircularEconomy" to "TransitionToACircularEconomy",
+            "PollutionPrevention" to "PollutionPreventionAndControl",
+            "Biodiversity" to "ProtectionAndRestorationOfBiodiversityAndEcosystems",
+        )
+        val criteriaObject = baseObject.opt("${criteriaType.prefix}Criteria") ?: return
+        if (criteriaObject is JSONObject) {
+            fieldRenamingHelper.keys.forEach { oldFieldName ->
+                if (criteriaObject.has(oldFieldName)) {
+                    val newKey = "${criteriaType.prefix}To${fieldRenamingHelper[oldFieldName]}"
+                    baseObject.put(
+                        if (criteriaType == CriteriaType.Dnsh) newKey else "${newKey}InPercent",
+                        criteriaObject.getOrJsonNull(oldFieldName),
+                    )
+                }
+            }
+        }
+        baseObject.remove("${criteriaType.prefix}Criteria")
+    }
+
+    private fun migrateFieldNames(baseObject: JSONObject) {
+        val fieldsToRemoveTotalPrefix = listOf(
+            "nonEligibleShare",
+            "eligibleShare",
+            "nonAlignedShare",
+            "alignedShare",
+            "enablingShare",
+            "transitionalShare",
+        )
+        val additionalFieldsToAddPercentIndication = listOf("enablingShare", "transitionalShare")
```

Comment:
Should those fields be called ...InPercent?

Reply:
No, not in the way the test is written. I chose to simply use the plain names of the fields as input here and handle the prefixes and suffixes below in the `forEach` lopp

Reply:
ok